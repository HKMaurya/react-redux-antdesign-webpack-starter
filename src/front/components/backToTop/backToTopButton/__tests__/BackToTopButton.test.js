// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import BackToTopButton from '../BackToTopButton';
// #endregion

describe('BackToTopButton component', () => {
  it('renders as expected', () => {
    const props = {
      position: null, // should be bottom-right by default
      onClick: () => {},
      motionStyle: {},
    };

    const component = renderer
      .create(
        <div>
          <BackToTopButton {...props}>
            <p>a child</p>
          </BackToTopButton>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
