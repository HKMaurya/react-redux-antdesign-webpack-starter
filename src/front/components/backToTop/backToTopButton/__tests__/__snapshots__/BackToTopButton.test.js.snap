// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTopButton component renders as expected 1`] = `
<div>
  <button
    className="btn"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#4A4A4A",
        "bottom": "40px",
        "left": "",
        "opacity": 0.5,
        "position": "fixed",
        "right": "-10px",
        "width": "100px",
        "zIndex": 10,
      }
    }
  >
    <p>
      a child
    </p>
  </button>
</div>
`;
